x/staking/module.go

```diff
@@ -99,6 +104,36 @@ type AppModule struct {
 	bankKeeper    types.BankKeeper
 }
 
+/////////////////////////////
```

Comment:
Could this be abstracted to it's own section?

Seems odd to me to put it in the `module.go`, any thoughts here?

Reply:
mmmm why?

Reply:
These functions represent genesis functionality and will only ever be called once, so maybe they should be inside the `genesis.go` file or maybe a new file `genesis_txs.go` where the interface [(defined in `cli/gentx.go`)](https://github.com/cosmos/cosmos-sdk/blob/paddy/generic-genutil-module/x/genutil/client/cli/gentx.go#L37-L42) could be satisfied then there would be a clear place where these types of genesis transactions live? Also, It would mean we don't need to add the extra complexity and lines to the `module.go` file. 🤔 

Although this may be premature optimization but IMO the implementation would be clearer and more defined.

wdyt?

Reply:
Yeah, I think more or less the same. It looks even for me something is not good when "appending it" into the module.

Even having another struct that keeps this methods and embedding it in the Module looks better for me that this one.

Reply:
I've moved this functionality to the genesis.go file, and defined a new type that satisfies the interface defined in the [gentx.go](https://github.com/cosmos/cosmos-sdk/blob/paddy/generic-genutil-module/x/genutil/client/cli/gentx.go#L37-L42) 

See here => https://github.com/cosmos/cosmos-sdk/pull/7279/commits/18dd9aa22d4534a7cb4eabdfc25aeb1c26bb5155